Clarify chghost

jwheare opened the pull request:
This PR replaces and builds on #364, incorporating the latest suggestions around showing unidented tildes, and also moving the fallback description to its own subheading.

jwheare commented:
I still need to do a fresh reading of these changes. Review comments from anyone else welcome.

edk0 commented on extensions/chghost-3.2.md:
```diff
@@ -5,59 +5,87 @@ copyrights:
-The CHGHOST message is one of the following:
+When the server sends a `CHGHOST` message to a client, it MUST defer doing so
+until both successful `NICK` and `USER` commands have been received by the
+server.  The server MAY choose to defer it until after registration is
```
I find this form (when you do X, you must defer doing X) pretty confusing.

jwheare replied:
Yeah I want to rewrite a bunch of this.

edk0 commented on extensions/chghost-3.2.md:
```diff
@@ -5,59 +5,87 @@ copyrights:
+shared channels, the simulated events SHOULD include appropriate `QUIT`, `JOIN`
+and `MODE` commands, to restore membership and user channel modes. For monitored
+clients, the events SHOULD include appropriate `RPL_MONOFFLINE` and
+`RPL_MONONLINE` numerics.
```
This seems questionable since it'll pointlessly highlight people.

jwheare replied:
Well, highlighting depends on the client. This allows clients that are monitoring users but aren't in shared channels to have an up to date mask, which is a requirement for various things.

jwheare replied:
Can't find much discussion of it other than these two comments:

https://github.com/ircv3/ircv3-specifications/pull/364#issuecomment-464385022
https://github.com/ircv3/ircv3-specifications/pull/364#pullrequestreview-204541454

Not sure which implementations if any do this. As a SHOULD it's not a breaking change, but it should definitely be errataed if kept, as you noted elsewhere.

jwheare replied:
From a glance at inspircd, unreal, charybdis, hybrid, none of these servers send monitor numerics in the fallback case, nor send the cap-enabled message to monitoring clients.

Given we aren't routinely extending MONITOR to support other forms of change notification (e.g. SETNAME) there's probably an argument for dropping it from this.

There's probably also an argument for extending MONITOR more thoroughly, but given lack of implementation, I reckon we can defer that.

Thoughts @SadieCat @syzop @slingamn?

edk0 commented on extensions/chghost-3.2.md:
```diff
@@ -5,59 +5,87 @@ copyrights:
+    :tim!b@ckyard CHGHOST toolshed backyard
 
 ## Errata
 
```
The requirement for monitored clients seems to be new, but isn't mentioned here.

jwheare commented:
I rewrote the intro and dropped the stuff about waiting till after NICK/USER. I think it's an edge case and don't think it's worth documenting in the spec. There are no known server implementations that still have an issue with this. Ref: https://github.com/ircv3/ircv3-specifications/pull/364#issuecomment-499463530

jwheare commented:
Will merge next week if no further comments.
